test/include/dlaf_test/blas/invoke.h

```diff
@@ -0,0 +1,61 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2021, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include "dlaf/types.h"
+
+#ifdef DLAF_WITH_CUDA
+#include <cublas_v2.h>
+#include <cuda_runtime.h>
+#include "dlaf/cublas/error.h"
+#include "dlaf/cuda/error.h"
+#endif
+
+namespace dlaf {
+namespace test {
+
+namespace internal {
+template <Device D>
+struct InvokeBlas;
+
+template <>
+struct InvokeBlas<Device::CPU> {
+  template <class F, class... Args>
+  static void call(F&& f, Args&&... args) {
+    f(std::forward<Args>(args)...);
+  }
+};
+
+#ifdef DLAF_WITH_CUDA
+template <>
+struct InvokeBlas<Device::GPU> {
+  template <class F, class... Args>
+  static void call(F&& f, Args&&... args) {
+    cublasHandle_t handle;
+    DLAF_CUBLAS_CALL(cublasCreate(&handle));
+  DLAF_CUBLAS_CALL(f(handle, std::forward<Args>(args)...);
+  DLAF_CUDA_CALL(cudaDeviceSynchronize());
+  DLAF_CUBLAS_CALL(cublasDestroy(handle));
```

Comment:
it seems that this has not been detected by the format check.

Reply:
Because the syntax is incorrect... I'm still working on the GPU part...